### PR TITLE
chore(iOS): multiple row selection support in ListView and tableview

### PR DESCRIPTION
**Pull request by vijaysingh-axway**
https://jira.appcelerator.org/browse/TIMOB-27142

### CODE REVIEW

File: apidoc/Titanium/UI/ListView.yml [hunk 1]
```diff
@@ -270,6 +270,21 @@ events:
         type: Boolean
         platforms: [iphone, ipad]
 
+  - name: rowsselected
```
Comment (hansemannn) [inline apidoc/Titanium/UI/ListView.yml]: List-views prefer the term "item" over "row". So either `itemsselected` or `multiselection`  would match better

File: iphone/Classes/TiUITableView.m [hunk 2]
```diff
@@ -2283,6 +2293,57 @@ - (void)tableView:(UITableView *)ourTableView moveRowAtIndexPath:(NSIndexPath *)
...
+
+      NSMutableDictionary *eventObject = [NSMutableDictionary dictionaryWithObjectsAndKeys:
+                                                                  section, @"section",
+                                                              NUMINT(dataIndex), @"index",
+                                                              row, @"row",
+                                                              row, @"rowData",
+                                                              nil];
+      [selectedItems addObject:eventObject];
+    }
+    [self.proxy fireEvent:@"rowsselected" withObject:@{ @"selectedRows" : selectedItems, @"startingRow" : startingRowObject }];
```
Comment (hansemannn) [inline iphone/Classes/TiUITableView.m]: This looks like exactly the code used in list-views. Maybe make it more generic?

File: iphone/Classes/TiUITableView.m [hunk 3]
```diff
@@ -2440,7 +2501,7 @@ - (void)keyboardDidChangeFrame:(NSNotification *)notification
 - (void)tableView:(UITableView *)ourTableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
 {
   BOOL search = NO;
-  if (!allowsSelectionSet || ![ourTableView allowsSelection]) {
+  if ((!allowsSelectionSet || ![ourTableView allowsSelection]) && !editing) {
```
Comment (hansemannn) [inline iphone/Classes/TiUITableView.m]: Are you sure this doesn't break apps that have `editing: true`?

File: apidoc/Titanium/UI/ListView.yml (hunk 1 above)
Comment (vijaysingh-axway) [reply apidoc/Titanium/UI/ListView.yml]: Agreed! Changed to 'itemsselected'.

File: iphone/Classes/TiUITableView.m (hunk 3 above)
Comment (vijaysingh-axway) [reply iphone/Classes/TiUITableView.m]: Yeah. I verified it. It's working fine for  me. If you have any test case please let me know. Thanks!

File: iphone/Classes/TiUITableView.m [hunk 4]
```diff
@@ -2053,6 +2053,16 @@ - (void)setContentInsets_:(id)value withObject:(id)props
   }
 }
 
+- (void)setAllowsMultipleSelectionDuringEditing_:(id)value
+{
+  ENSURE_TYPE(value, NSNumber);
+  [[self proxy] replaceValue:value forKey:@"allowsMultipleSelectionDuringEditing" notification:NO];
+
+  [[self tableView] beginUpdates];
```
Comment (janvennemann) [inline iphone/Classes/TiUITableView.m]: Why do we need `beginUpdates` and `endUpdates` here? We are not modifying rows/sections of the table view, are we?

File: iphone/Classes/TiUITableView.m [hunk 5]
```diff
@@ -2283,6 +2293,57 @@ - (void)tableView:(UITableView *)ourTableView moveRowAtIndexPath:(NSIndexPath *)
...
+      NSInteger sectionIdx = [index section];
+      NSArray *sections = [(TiUITableViewProxy *)[self proxy] internalSections];
+      TiUITableViewSectionProxy *section = [self sectionForIndex:sectionIdx];
+
+      NSInteger rowIndex = [index row];
+      int dataIndex = 0;
+      int c = 0;
+      TiUITableViewRowProxy *row = [section rowAtIndex:rowIndex];
+
+      for (TiUITableViewSectionProxy *section in sections) {
```
Comment (janvennemann) [inline iphone/Classes/TiUITableView.m]: This for loop get's used a lot in our table view proxy. Make a function out of it so we can reuse the code.

**Review by janvennemann — CHANGES_REQUESTED** [review]

File: iphone/Classes/TiUITableView.m (hunk 5 above)
Comment (vijaysingh-axway) [reply iphone/Classes/TiUITableView.m]: Updated

**Comment by vijaysingh-axway** [comment]
@janvennemann Updated PR as per review comment. Please review again. Thanks!

**Review by janvennemann — APPROVED** [review]
Thanks, looking good now, approved!

**Comment by keerthi1032** [comment]
FR passed. Able to select multiple row on iOS 13 

Test Enviornment:
Name                        = Mac OS X
  Version                     = 10.14.5
  Architecture                = 64bit
  Memory                      = 17179869184
Node.js
  Node.js Version             = 10.16.2
  npm Version                 = 6.9.0
Titanium CLI
  CLI Version                 = 5.2.1
Titanium SDK
  SDK Version                 = local sdk 8.2.0.v20190805105616
Device - iPhone XR iOS 13 beta 6
Simulator -iPhone X iOS 13
CLi -7.1.1 -master 2
Studio 5.1.3.201907112159

**Pull request merged**